dvc/stages/__init__.py

```diff
@@ -726,33 +548,7 @@ def dumpd(self):
         }
 
     def dump(self):
-        fname = self.path
-
-        self._check_dvc_filename(fname)
-
-        logger.debug(
-            "Saving information to '{file}'.".format(file=relpath(fname))
-        )
-        state = self.dumpd()
-
-        # When we load a stage we parse yaml with a fast parser, which strips
-        # off all the comments and formatting. To retain those on update we do
-        # a trick here:
-        # - reparse the same yaml text with a slow but smart ruamel yaml parser
-        # - apply changes to a returned structure
-        # - serialize it
-        if self._stage_text is not None:
-            saved_state = parse_stage_for_update(self._stage_text, fname)
-            # Stage doesn't work with meta in any way, so .dumpd() doesn't
-            # have it. We simply copy it over.
-            if "meta" in saved_state:
-                state["meta"] = saved_state["meta"]
-            apply_diff(state, saved_state)
-            state = saved_state
-
-        dump_stage_file(fname, state)
-
-        self.repo.scm.track_file(relpath(fname))
+        self.dvcfile.dump_single_stage(self)
```

Comment:
Discussed with @skshetry that this is a bit backwards and it makes sense to get rid of `dump` method in Stage and just use Dvcfile.dump, to simplify next steps.